test/parallel_api/algorithm/alg.sorting/alg.set.operations/set.pass.cpp

```diff
@@ -13,357 +13,7 @@
-            else
-                invoke_on_all_policies<4>()(test_set_union<T1>(), in1.begin(), in1.end(), in2.cbegin(), in2.cend());
-#endif
-
-#ifdef _PSTL_TEST_SET_INTERSECTION
-            if(comp_flag)
-                invoke_on_all_policies<1>()(test_set_intersection<T1>(), in1.begin(), in1.end(), in2.cbegin(), in2.cend(),
-                                            compare);
-            else
-                invoke_on_all_policies<5>()(test_set_intersection<T1>(), in1.begin(), in1.end(), in2.cbegin(), in2.cend());
-#endif
-#ifdef _PSTL_TEST_SET_DIFFERENCE
-            if(comp_flag)
-                invoke_on_all_policies<2>()(test_set_difference<T1>(), in1.begin(), in1.end(), in2.cbegin(), in2.cend(),
-                                            compare);
-            else
-                invoke_on_all_policies<6>()(test_set_difference<T1>(), in1.begin(), in1.end(), in2.cbegin(), in2.cend());
-#endif
-#ifdef _PSTL_TEST_SET_SYMMETRIC_DIFFERENCE
-            if(comp_flag)
-                invoke_on_all_policies<3>()(test_set_symmetric_difference<T1>(), in1.begin(), in1.end(), in2.cbegin(),
-                                            in2.cend(), compare);
-            else
-                invoke_on_all_policies<7>()(test_set_symmetric_difference<T1>(), in1.begin(), in1.end(), in2.cbegin(),
-                                                in2.cend());
-#endif
-        }
-    }
-}
-
-template <typename T>
-struct test_non_const_set_difference
-{
-    template <typename Policy, typename InputIterator, typename OutputInterator>
-    void
-    operator()(Policy&& exec, InputIterator input_iter, OutputInterator out_iter)
-    {
-        set_difference(exec, input_iter, input_iter, input_iter, input_iter, out_iter, non_const(::std::less<T>()));
-    }
-};
-
-template <typename T>
-struct test_non_const_set_intersection
-{
-    template <typename Policy, typename InputIterator, typename OutputInterator>
-    void
-    operator()(Policy&& exec, InputIterator input_iter, OutputInterator out_iter)
-    {
-        set_intersection(exec, input_iter, input_iter, input_iter, input_iter, out_iter, non_const(::std::less<T>()));
-    }
-};
-
-template <typename T>
-struct test_non_const_set_symmetric_difference
-{
-    template <typename Policy, typename InputIterator, typename OutputInterator>
-    void
-    operator()(Policy&& exec, InputIterator input_iter, OutputInterator out_iter)
-    {
-        set_symmetric_difference(exec, input_iter, input_iter, input_iter, input_iter, out_iter,
-                                 non_const(::std::less<T>()));
-    }
-};
-
-template <typename T>
-struct test_non_const_set_union
-{
-    template <typename Policy, typename InputIterator, typename OutputInterator>
-    void
-    operator()(Policy&& exec, InputIterator input_iter, OutputInterator out_iter)
-    {
-        set_union(exec, input_iter, input_iter, input_iter, input_iter, out_iter, non_const(::std::less<T>()));
-    }
-};
+#include "set.pass.h"
```

Comment:
There are no files named with "*.pass.h" pattern. Could you remove "pass" from the name?

Reply:
In this case we will have file with name ```set.h```, it's too much from my point of view :-)

Reply:
I see your point. There might some clash with the header names (I guess you meant that). What do you think about set_utils.h?

Reply:
Or "set_common.h"?

Reply:
Fixed